Simulation Data

ohsnapitscolin opened the pull request:
Adds a new data structure for the `simulation` subchapter type. This controls the `min`, `max,` `step` and `defaultValue` values for all of the inputs, as well as the `label` and `tooltip` text.

The page title, explanation and image have been moved into this structure as well. Additionally the structure supports `sections`, which can move used for any simulations that have multiple tabs. This can be demo'd under the `"Testing" tabs under the second subchapter.

Also added a "Reset" button to prove out the logic for reseting the inputs.

This PR also adds the `Slider` component, which support showing the default value and the current value under the thumb.

Styling not finalized.

zscheinfeld commented:
Wow the tab structure is awesome!! Can't wait to style this

zscheinfeld commented:
I squashed and merged this and am now noticing a few small things. 

1.It looks like the subchapters are not changing from 3.1 to 3.2, and I'm no longer seeing the tabs. Not sure if the squash and merge is what caused this?

2. When you transition to fine print, the sliders stay. I think this is a z-index thing we can quickly fix. 

Thanks again!

ohsnapitscolin commented:
> @ohsnapitscolin I squashed and merged this and am now noticing a few small things:
> 
> 1.It looks like the subchapters are not changing from 3.1 to 3.2, and I'm no longer seeing the tabs. Not sure if the squash and merge is what caused this?
> 
> 2. When you transition to fine print, the sliders stay. I think this is a z-index thing we can quickly fix.
> 
> Thanks again!

@zscheinfeld got some small fixes [here](https://github.com/zscheinfeld/silverlinings/pull/7). Updated the simulation page to just be a scroll for now while we figure out exactly how we want to handle the styles there.
